src/getgb2r.f

```diff
@@ -1,197 +1,159 @@
+C>    @file
+C>    @brief This subroutine read and unpack sections 6 and 7 from ah
+C>    grib2 message.
+C>    @author Stephen Gilbert @date 2002-01-11
+C>
+
+C>    This subroutine read and unpack sections 6 and 7 from a grib2
+C>    message. It assumes that the "metadata" for this field already
+C>    exists in derived type gribfield. Specifically, it requires
+C>    gfld\%ibmap, gfld\%ngrdpts, gfld\%idrtnum, gfld\%idrtmpl, and
+C>    gfld\%ndpts. It decoded information for the selected grib field and
+C>    returned it in a derived type variable, gfld. gfld is of type
+C>    gribfield, which is defined in module grib_mod, so users of this
+C>    routine will need to include the line "use grib_mod" in their
+C>    calling routine. Each component of the gribfield type is described
+C>    in the output argument list section below.
+C>
+C>    @param[in] LUGB integer unit of the unblocked grib data file.
+C>    @param[in] CINDEX index record of the grib field (see docblock of
+C>    subroutine ixgb2 for description of an index record.)
+C>    @param[out] GFLD derived type gribfield (defined in module grib_mod)
+C>    (NOTE: See Remarks Section)
+C>    - gfld\%version GRIB edition number (currently 2)
+C>    - gfld\%discipline Message Discipline (see Code Table 0.0)
+C>    - gfld\%idsect Contains the entries in the Identification Section
+C>    (Section 1) This element is actually a pointer to an array
+C>    that holds the data.
+C>    - gfld\%idsect(1) Identification of originating Centre
+C>    (see Common Code Table C-1) 7 US National Weather Service
+C>    - gfld\%idsect(2) Identification of originating Sub-centre
+C>    - gfld\%idsect(3) GRIB Master Tables Version Number
+C>    (see Code Table 1.0) 0 Experimental; 1 Initial operational version number
+C>    - gfld\%idsect(4) GRIB Local Tables Version Number (see Code Table 1.1)
+C>    0 Local tables not used; 1-254 Number of local tables version used
+C>    - gfld\%idsect(5) Significance of Reference Time (Code Table 1.2)
+C>    0 Analysis; 1 Start of forecast; 2 Verifying time of forecast;
+C>    3 Observation time.
+C>    - gfld\%idsect(6) Year (4 digits)
+C>    - gfld\%idsect(7) Month
+C>    - gfld\%idsect(8) Day
+C>    - gfld\%idsect(9) Hour
+C>    - gfld\%idsect(10) Minute
+C>    - gfld\%idsect(11) Second
+C>    - gfld\%idsect(12) Production status of processed data (see Code
+C>    Table 1.3)
+C>    0 Operational products; 1 Operational test products;
+C>    2 Research products; 3 Re-analysis products.
+C>    - gfld\%idsect(13) Type of processed data (see Code Table 1.4)
+C>    0 Analysis products
```

Comment:
You should make these sub-lists, like this:
```
C>    - gfld\%idsect(13) Type of processed data (see Code Table 1.4)
C>      - 0 Analysis products
C>      - 1 Forecast products
C>      - 2 Analysis and forecast products
C>      - 3 Control forecast products

```

Reply:
changed and added